jmespath-core/src/main/java/io/burt/jmespath/BaseAdapter.java

```diff
@@ -60,8 +58,8 @@ public int compare(T value1, T value2) {
         case BOOLEAN:
           return (isTruthy(value1) && isTruthy(value2)) || (!isTruthy(value1) && !isTruthy(value2)) ? 0 : -1;
         case NUMBER:
-          Comparable d1 = toNumber(value1).doubleValue();
-          Comparable d2 = toNumber(value2).doubleValue();
+          Double d1 = toNumber(value1).doubleValue();
+          Double d2 = toNumber(value2).doubleValue();
           return d1.compareTo(d2);
```

Comment:
I'm not sure whether JMESPath is supposed to support comparing Booleans or not. I was referring to the JavaDoc comment for the method, which implied that Booleans should be comparable, when in fact they are not.

Assuming Booleans shouldn't be comparable in JMESPath, I'm fine with the current logic (although it could be simplified to `(isTruthy(value1) == isTruthy(value2)) ? 0 : -1`), but then the JavaDoc comment should be updated accordingly.